pkg/cli/admin/prune/images/images.go

```diff
@@ -388,20 +386,43 @@ func (o PruneImagesOptions) Run() error {
 		limitRangesMap[limit.Namespace] = limits
 	}
 
+	imageStreamWatcher, err := o.ImageClient.ImageStreams(o.Namespace).Watch(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("internal error: failed to watch for image streams: %v"+
+			"\n - image stream changes will not be detected", err))
+		imageStreamWatcher = watch.NewFake()
+	}
+	defer imageStreamWatcher.Stop()
+
+	allStreams, err := o.ImageClient.ImageStreams(o.Namespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	allStreamsMap := map[string]*imagev1.ImageStream{}
+	for i := range allStreams.Items {
+		stream := &allStreams.Items[i]
+		allStreamsMap[fmt.Sprintf("%s/%s", stream.Namespace, stream.Name)] = stream
+	}
+
 	ctx := context.TODO()
 	allImagesUntyped, _, err := pager.New(func(ctx context.Context, opts metav1.ListOptions) (runtime.Object, error) {
 		return o.ImageClient.Images().List(context.TODO(), opts)
-	}).List(ctx, metav1.ListOptions{})
+	}).List(ctx, metav1.ListOptions{
+		Limit: 5000,
+	})
 	if err != nil {
 		return err
 	}
-	allImages := &imagev1.ImageList{}
+	allImages := map[string]*imagev1.Image{}
 	if err := meta.EachListItem(allImagesUntyped, func(obj runtime.Object) error {
-		allImages.Items = append(allImages.Items, *obj.(*imagev1.Image))
+		image := obj.(*imagev1.Image)
+		allImages[image.Name] = image
 		return nil
 	}); err != nil {
 		return err
 	}
+	allImagesUntyped = nil // We don't need it anymore
```

Comment:
cool, I didn't know about EachListItem, fixed.

btw, the example that you gave me also needs to be updated. SimplePageFunc is needed for list functions that don't accept ctx. And the list function uses context.TODO(), but it could use `ctx` from the pager.